ros-tutorials/lesson-3-write-a-node.md

```diff
@@ -3,29 +3,35 @@
+
+This final tutorial will not necessarily provide all the information necessary. This is by design. The intention is to teach problem solving in the situation when you know _what_ you want to do, but not _how_ to do it. If there are things you do not recognize or feel comfortable with, take initiative and look them up online. Read tutorials, documentation, etc.
+
+:warning: Moving forward, you will need to program in Python. If you do not have experience coding, you will need to learn Python. Find a tutorial/course you like online, or pick one from [here](https://stackify.com/learn-python-tutorials/#post-21937-_u49dnodg9ai6) or [here](https://gitconnected.com/learn/python). If you have experience coding in another language, feel free to proceed, as Python should be fairly easy for you to pick up.
 
 ## Vocab
-- Package: A folder of nodes, messages, all related to some common goal or concept
-- `catkin`: The package managing system. This lets you create a workspace, create packages, manage packages, etc.
+- **Package**: A folder of nodes, messages, and other files all related to some common goal
 
 ## Goals
-1. Create your own package
-   - Be sure to use the newer versions of catkin commands (see Catkin Tools below). For example, do not use `catkin_make`, use `catkin build`. Similarly, `catkin_create_pkg` becomes `catkin create pkg`.
-2. Create a new node inside that package
-3. Write this node to make the turtle move in a square
-   - Hint: You will need to publish something somewhere
+1. Program a node inside that package by editing the premade Python file.
+    - Hint: You will need to publish something somewhere.
+2. Run this node to make the turtle move in a square.
 
 ## Resources
-- [Packages](http://wiki.ros.org/ROS/Tutorials/CreatingPackage)
+- [turtlesim](http://wiki.ros.org/turtlesim)
 - [Publisher & Subscriber](http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28python%29)
-- [Catkin Tools](https://catkin-tools.readthedocs.io/en/latest/index.html)
-   - [Quickstart](https://catkin-tools.readthedocs.io/en/latest/quick_start.html) to initialize new workspace
+- [Video tutorial #1](https://youtu.be/4Lifb9Cg_9w) (extra help)
+- [Video tutorial #2](https://youtu.be/D7ISrmszozk) (extra help)
 
 ## Notes
-- You may want to run the docker container using a mount version of the `docker run` command to save your files (remember the [Docker Intro](https://github.com/DukeRobotics/documentation/blob/master/docker/README.md))
-- You will not be able to use the code in the 'Publisher and Subscriber' tutorial exactly as it is
+- The starter code's premade package is found in the folder you cloned in Lesson 0, here: `documentation/ros-tutorials/catkin_ws`. There is a Python file in there for you to edit.
+    - Inside the container, you can find the package at `~/dev/catkin_ws`.
+    - How does that work? The docker-compose command we had you run in Lesson 0 "mounted" the starter code that is on your computer into the container.
```

Comment:
How does what work? Maybe explicitly say that code changes on your local computer will reflect in the container.

Reply:
Thanks, clarified this.